communote/webapp/src/main/webapp/javascript/utils/BinaryDataPasteListener.js

```diff
@@ -0,0 +1,170 @@
+            }
+            typeRegex.lastIndex = 0;
+        }
+        return true;
+    }
+
+    /**
+     * Create a listener for the 'paste' event and attach it to the given element. Whenever the
+     * paste event is fired and the pasted data is binary the given callback is invoked with the
+     * pasted data. If the browser doesn't support processing of pasted binary data, no listener
+     * will be attached to the element. Can for instance be used to handle paste of clipboard image
+     * data.
+     * 
+     * @param {(String|Element)} elem - CSS selector of the element or the element to attach to. If
+     *            the element is undefined or the selector does not return an element an error is
+     *            thrown.
+     * @param {Function} callback - the function to call for a paste event with binary data. The
+     *            function will be passed an array of objects where each object represents a pasted
+     *            binary blob and has the members blob and fileName. The blob holds the data and is
+     *            an instance of Blob. The fileName is a string with the name of the blob as
+     *            provided by the browser or constructed from the fileNamePrefix option if defined.
+     * @param {?Object} options - optional options for customizing the listener
+     * @param {?String} options.fileNamePrefix - prefix to use for creating a unique filename for
+     *            each pasted binary data. If provided this name will be passed to the callback
+     *            instead of the one given by the browser.
+     * @param {?String} options.fileNameDefaultExtension - file extension to append to the unique
+     *            filename if the name provided by the browser has no extension. Will be ignored if
+     *            fileNamePrefix option is not set.
+     * @param {?boolean} options.preventDefault - prevent default handling after processing the
+     *            paste event. Defaults to true.
+     * @param {?boolean} options.stopPropagation - stop propagation after processing the paste
+     *            event. Defaults to true.
+     * @param {?RegExp} options.typeRegex - regular expression to apply on the mime type of the
+     *            pasted data to only handle binary data with a matching mime type. If not given, all
+     *            binary data is handled. If a captured paste event has no binary data matching the
+     *            regex, the propagation or default handling of the event won't be stopped or
+     *            prevented.
+     * 
+     * @class
+     */
+    var Listener = function(elem, callback, options) {
+        if (typeof elem === 'string') {
+            elem = document.querySelector(elem);
+        }
+        if (!elem || !elem.addEventListener) {
+            throw new Error('No element provided for adding the paste event listener');
+        }
+        // ignore if callback is missing or not supported
+        if (callback && binaryPasteSupported) {
```

Comment:
Diese Prüfung auf binaryPasteSupport an dieser Stelle klingt so, als sollte man es ganz am Anfang prüfen und nur dann die funktionen dafür registrieren. dann bräuchte man es später nicht mehr prüfen

Reply:
das ist gewollt. Die Idee ist, dass man die Klasse unabhängig vom Browser-Support bereitstellt. Dadurch muss der Entwickler, der diesen Listener verwenden will, keine Weichen in seinen Code einbauen. Er kann einfach den Konstruktor aufrufen und bekommt eine Instanz zurück, deren Methoden er unabhängig vom Browser-Support aufrufen kann. Da passiert dann halt einfach nix, wenn die Browserunterstützung fehlt.